src/Symfony/Component/Security/Http/Firewall/DigestAuthenticationListener.php

```diff
@@ -78,7 +78,7 @@ public function handle(GetResponseEvent $event)
         }
 
         try {
-            $digestAuth->validateAndDecode($this->authenticationEntryPoint->getKey(), $this->authenticationEntryPoint->getRealmName());
+            $digestAuth->validateAndDecode($this->authenticationEntryPoint->getSecret(), $this->authenticationEntryPoint->getRealmName());
```

Comment:
We do not seem to have tests covering this. But they would need to be added in older versions.